File: src/maasserver/static/js/angular/factories/ipranges.js
/* Copyright 2016 Canonical Ltd.  This software is licensed under the
 * GNU Affero General Public License version 3 (see the file LICENSE).
 *
 * MAAS IPRange Manager
 *
 * Manages all of the IPRanges in the browser. The manager uses the
 * RegionConnection to load the IPRanges, update the IPRanges, and listen for
 * notification events about IPRanges.
 */

function IPRangesManager(RegionConnection, Manager) {
  function IPRangesManager() {
    Manager.call(this);

    this._pk = "id";
    this._handler = "iprange";

    // Listen for notify events for the iprange object.
    var self = this;
    RegionConnection.registerNotifier("iprange", function(action, data) {
      self.onNotify(action, data);
    });
  }

  IPRangesManager.prototype = new Manager();

  return new IPRangesManager();
}

IPRangesManager.$inject = ["RegionConnection", "Manager"];

export default IPRangesManager;

File: src/maasservicelayer/models/ipranges.py
# Copyright 2024-2025 Canonical Ltd.  This software is licensed under the
# GNU Affero General Public License version 3 (see the file LICENSE).


from pydantic import IPvAnyAddress

from maascommon.enums.ipranges import IPRangeType
from maasservicelayer.models.base import (
    generate_builder,
    MaasTimestampedBaseModel,
)


@generate_builder()
class IPRange(MaasTimestampedBaseModel):
    type: IPRangeType
    start_ip: IPvAnyAddress
    end_ip: IPvAnyAddress
    comment: str | None = None
    subnet_id: int
    user_id: int | None = None
